### ISSUE TITLE
Wrong service account is used if a project ID has a prefix for its org

### ISSUE THREAD

**Issue by zhenghuiwang** [kubeflow/kubeflow#2244]
My project with id `google.com:codelab-gae-blog-golang`, the service account generated by the

`scripts/kfctl.sh generate platform && scripts/kfctl.sh apply platform ` are

```
$ gcloud iam service-accounts list   
                                                                                                                       
NAME                                                                       EMAIL
App Engine default service account                                         codelab-gae-blog-golang.google.com@appspot.gserviceaccount.com
kubeflow service account                                                   kubeflow-deploy-admin@codelab-gae-blog-golang.google.com.iam.gserviceaccount.com
Compute Engine default service account                                     571461128879-compute@developer.gserviceaccount.com
Service Account used for Kubeflow admin actions.                           zhenghui-kubeflow-admin@codelab-gae-blog-golang.google.com.iam.gserviceaccount.com
Service Account used for Kubeflow user actions.                            zhenghui-kubeflow-user@codelab-gae-blog-golang.google.com.iam.gserviceaccount.com
GCP Service Account to use as VM Service Account for Kubeflow Cluster VMs  zhenghui-kubeflow-vm@codelab-gae-blog-golang.google.com.iam.gserviceaccount.com
```
But the I get the following error:

```
ERROR: (gcloud.deployment-manager.deployments.update) Error in Operation [operation-1546992226734-57efb32250fb0-faa71f47-e342fdf2]: errors:                                                                                            
- code: RESOURCE_ERROR                                                                                                                                                                                                                 
  location: /deployments/zhenghui-kubeflow/resources/zhenghui-kubeflow                                                                                                                                                                 
  message: '{"ResourceType":"container.v1.cluster","ResourceErrorCode":"400","ResourceErrorMessage":{"code":400,"message":"Service                                                                                                     
    account \"zhenghui-kubeflow-vm@google.com:codelab-gae-blog-golang.iam.gserviceaccount.com\"                                                                                                                                        
    is not valid.","status":"INVALID_ARGUMENT","statusMessage":"Bad Request","requestPath":"https://container.googleapis.com/v1/projects/google.com:codelab-gae-blog-golang/zones/us-west1-a/clusters","httpMethod":"POST"}}' 
````

Notice the service account used in the script is `zhenghui-kubeflow-vm@google.com:codelab-gae-blog-golang.iam.gserviceaccount.com` not what has been created `zhenghui-kubeflow-vm@codelab-gae-blog-golang.iam.gserviceaccount.com`

**Comment by jlewi** [comment]
@zhenghuiwang Did you specify your project as `google.com:codelab-gae-blog-golang` or as just `codelab-gae-blog-golang` ?

We regularly use google.com and kubeflow.org projects and I believe it works. I think we specify the project without the org and it works.

I suspect if you specify the project and include the domain that might cause the problem.

**Comment by zhenghuiwang** [comment]
I use project ID `google.com:codelab-gae-blog-golang` which is showed in the GCP web console. I remember I tried to use `codelab-gae-blog-golang` as the project ID but some other steps broke.

The project was created in December 2018. I created some projects in January and they don't have the org prefix. So the problem is not reproducible for new projects.

I'm closing it because it seems no one reported the same issue and it won't happen for new projects.
